src/accelerate/accelerator.py

```diff
@@ -2299,6 +2308,124 @@ def save(self, obj, f):
         """
         save(obj, f)
 
+    def save_model(
+        self,
+        model: torch.nn.Module,
+        save_directory: Union[str, os.PathLike],
+        max_shard_size: Union[int, str] = "10GB",
+        safe_serialization: bool = False,
+    ):
+        """
+        Save a model so that it can be re-loaded using load_checkpoint_in_model
+
+        Arguments:
+            model: (`torch.nn.Module`):
+                Model to be saved. The model can be wrapped or unwraped.
+            save_directory (`str` or `os.PathLike`):
+                Directory to which to save. Will be created if it doesn't exist.
+            max_shard_size (`int` or `str`, *optional*, defaults to `"10GB"`):
+                The maximum size for a checkpoint before being sharded. Checkpoints shard will then be each of size
+                lower than this size. If expressed as a string, needs to be digits followed by a unit (like `"5MB"`).
+
+                <Tip warning={true}>
+
+                If a single weight of the model is bigger than `max_shard_size`, it will be in its own checkpoint shard
+                which will be bigger than `max_shard_size`.
+
+                </Tip>
+
+            safe_serialization (`bool`, *optional*, defaults to `False`):
+                Whether to save the model using `safetensors` or the traditional PyTorch way (that uses `pickle`).
+        """
+
+        if safe_serialization and not is_safetensors_available():
+            raise ImportError("`safe_serialization` requires the `safetensors library: `pip install safetensors`.")
+
+        if os.path.isfile(save_directory):
+            logger.error(f"Provided path ({save_directory}) should be a directory, not a file")
+            return
+
+        # unwrap model
+        model_to_save = self.unwrap_model(model)
```

Comment:
This won't be necessary since `self.get_state_dict` will do it.

Reply:
Oups ;)